### PR TITLE
(fix) AnalyzerEbur128: catch invalid return values inf/-inf

### PR DESCRIPTION
**Pull request by ronso0**
see #13676

### CODE REVIEW

File: src/util/versionstore.cpp [hunk 1]
```diff
@@ -151,6 +152,10 @@ QStringList VersionStore::dependencyVersions() {
     sf_command(nullptr, SFC_GET_LIB_VERSION, sndfile_version, sizeof(sndfile_version));
     // Null-terminate just in case.
     sndfile_version[sizeof(sndfile_version) - 1] = '\0';
+
+    int eburMaj, eburMin, eburP = 0;
```
Comment (Swiftb0y) [inline src/util/versionstore.cpp:156]: this leaves the first two variables uninitialized, which is one of the key reason why this style is discouraged.
```suggestion
    int eburMaj = 0;
    int eburMin = 0; 
    int eburP = 0;
```

**Review by Swiftb0y — CHANGES_REQUESTED** [review]

File: src/util/versionstore.cpp (hunk 1 above)
Comment (ronso0) [reply src/util/versionstore.cpp]: thanks, I pushed a fixup.

**Review by Swiftb0y — APPROVED** [review]
LGTM. Thank you. How reproducible is the issue? We should escalate this to upstream libebur128.

**Comment by ronso0** [comment]
I can reproduce it with a silent track (empty Mixxx recording, ~3 sec)
[space.mp3.zip](https://github.com/user-attachments/files/17085084/space.mp3.zip)

**Comment by Swiftb0y** [comment]
right, makes sense that you'd need "infinite" gain to get that to a fixed level. This should be caught but libebur though.

File: src/analyzer/analyzerebur128.cpp [hunk 2]
```diff
@@ -69,7 +69,13 @@ void AnalyzerEbur128::storeResults(TrackPointer pTrack) {
         qWarning() << "AnalyzerEbur128::storeResults() failed with" << e;
         return;
     }
-    if (averageLufs == -HUGE_VAL || averageLufs == 0.0) {
+    if (averageLufs == -HUGE_VAL ||
+            averageLufs == HUGE_VAL ||
+            // This catches inf and -inf that slipped through in libebur for some reason..
+            // TODO Move this to a VERIFY_OR_DEBUG_ASSERT() once the root cause
+            // has been identified.
```
Comment (daschuer) [inline src/analyzer/analyzerebur128.cpp:76]:
```suggestion
            // This catches inf and -inf that slipped through in libebur for some reason.
            // TODO: Move this to a VERIFY_OR_DEBUG_ASSERT() once the root cause
            // has been identified.
```

File: src/analyzer/analyzerebur128.cpp [hunk 3]
```diff
@@ -69,7 +69,13 @@ void AnalyzerEbur128::storeResults(TrackPointer pTrack) {
...
+            util_isinf(averageLufs) != 0 ||
```
Comment (daschuer) [inline src/analyzer/analyzerebur128.cpp:77]: util_isinf introduces a insymmetry compared the the C89 HUGE_VAL constant. 

How about this:
```suggestion
            util_isnormal(averageLufs) ||
```
We can only really deal with a normal value. 
The alternative would be to check for HUGE_VAL instead. 

There is by the way the issue that util_isnormal() shall return bool, not int.

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/analyzer/analyzerebur128.cpp (hunk 3 above)
Comment (ronso0) [reply src/analyzer/analyzerebur128.cpp:77]: Thanks, I've prepended a commit to fix the return types in fpclassify and use `util_isnormal()` here

**Comment by ronso0** [comment]
> How reproducible is the issue? We should escalate this to upstream libebur128.

Hmm, the last (code) activity was 3 years ago.

File: src/analyzer/analyzerebur128.cpp [hunk 4]
```diff
@@ -69,7 +69,13 @@ void AnalyzerEbur128::storeResults(TrackPointer pTrack) {
...
         return;
     }
-    if (averageLufs == -HUGE_VAL || averageLufs == 0.0) {
+    if (averageLufs == -HUGE_VAL ||
+            averageLufs == HUGE_VAL ||
+            // This catches 0, inf and -inf (that may have slipped through in
+            // libebur for some reason.
+            // TODO Move this to a VERIFY_OR_DEBUG_ASSERT() once the root cause
+            // has been identified, and only do a zero check here.
+            util_isnormal(averageLufs)) {
```
Comment (daschuer) [inline src/analyzer/analyzerebur128.cpp:78]: Now the check for 0 is gone. 
My idea was to just

```suggestion
    if (!util_isnormal(averageLufs) || averageLufs == 0.0) {
            // This catches 0, inf (HUGE_VAL), -inf (-HUGE_VAL) and denormals (treated as zero), 
            // that may have slipped through in libebur for some reason.
```

Returning -HUGE_VAL is the documented invalid value. I am in doubt there will be a "fix" for HUGE_VAL and even then we deal with user data, where Bad things are not "impossible". So I would not add the TODO.

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/analyzer/analyzerebur128.cpp (hunk 4 above)
Comment (ronso0) [reply src/analyzer/analyzerebur128.cpp:78]: Zero is covered by std::isnormal
https://en.cppreference.com/w/cpp/numeric/math/isnormal

I'll remove the TODO then.

Comment (daschuer) [reply src/analyzer/analyzerebur128.cpp:78]: Ups, thank you for the hint.

**Review by daschuer — APPROVED** [review]
LGTM, thank you.

**Pull request merged**